### PR TITLE
devtools: api to inspect service worker

### LINKED ISSUE #1331: API to inspect Service Workers
One of the things that will be super useful for Desktop applications that build on top of remote content is the Shared [Service Workers API](http://www.html5rocks.com/en/tutorials/service-worker/introduction/), which basically lets you have a programmable "proxy" for network requests at a high-level and trump normal browser caching rules. With this, you can write offline apps without having to ship all of your app locally / your update story is way saner, etc etc etc.

This is super cool, but without `chrome://serviceworker-internals` it's *way* harder (read: impossible) to know what's going on (if your list is empty, visit https://events.google.com/io2015 a few times). 

To that end, I propose a few additions to the WebContents API:

* `hasServiceWorker` - determine if this page has a registration for a service worker
* `unregisterServiceWorker` - Kill any registration for a service worker if it exists
* `openServiceWorkerDevTools` - Open DevTools for the service worker associated with this page

Thoughts?

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #1331 

Depends on https://github.com/atom/brightray/pull/107

- [x] Need to implement for browser window (facing some crashes), currently works for webview

### CODE REVIEW

**Comment by deepak1556** [comment]
with next update for chromium we get better observers for [service wokers](https://code.google.com/p/chromium/codesearch#chromium/src/content/browser/devtools/service_worker_devtools_manager.h&sq=package:chromium&rcl=1431934741&l=30)

**Comment by paulcbetts** [comment]
Into it!

File: atom/browser/native_window.cc [hunk 1]
```diff
@@ -317,6 +317,17 @@ void NativeWindow::InspectElement(int x, int y) {
   agent->InspectElement(x, y);
 }
 
+void NativeWindow::InspectServiceWorker() {
```
Comment (zcbenz) [inline atom/browser/native_window.cc]: Why adding `InspectServiceWorker` for `NativeWindow`?

Comment (deepak1556) [inline atom/browser/native_window.cc]: to support `BrowserWindow.inspectServiceWorker` .

Comment (zcbenz) [inline atom/browser/native_window.cc]: Why is `BrowserWindow.inspectServiceWorker` needed? I think we can just use `window.webContents.inspectServiceWorker`.

Comment (deepak1556) [inline atom/browser/native_window.cc]: All the devtools api we have listed are on `BrowserWindow` besides if they use `window.webcontents` before its finished loading wouldnt `webcontents.opendevtools` crash ?

Comment (zcbenz) [inline atom/browser/native_window.cc]: :+1:

**Pull request merged**